sources/packages/backend/libs/integrations/src/services/sin-validation/sin-validation.service.ts

```diff
@@ -74,120 +80,190 @@ export class SINValidationService extends RecordDataModelService<SINValidation>
-      // The SIN validation is already present and the received status is different from
-      // the existing one. It can happen when the status received is 2 (under review) and later
-      // a new status will be received once the review is done, what means that one request can
-      // result in more than one response. In this case, if the received record is the most updated
-      // for the student, a "cloned" record will be inserted.
-      const mostUpdatedRecordDate = await this.repo
-        .createQueryBuilder("sinValidation")
-        .select("MAX(sinValidation.dateReceived)", "maxDate")
-        .innerJoin("sinValidation.student", "student")
-        .where("student.id = :student", {
-          student: existingValidation.student.id,
-        })
-        .getRawOne();
-      if (processDate > mostUpdatedRecordDate.maxDate) {
-        // This will force the record to be inserted with all the values selected
-        // during the initial SQL query.
-        delete existingValidation.id;
-        // Insert the new SIN validation and associate it with the student
-        // as the most updated one to be used as a current.
-        await this.studentService.updateSINValidationByStudentId(
-          existingValidation.student.id,
-          existingValidation,
-          auditUserId,
-        );
-        operationDescription = `Created a new SIN validation because the record id is already present and updated.`;
-        const clonedRecord = await this.repo.save(existingValidation);
+        if (sinStatusChanged) {
+          // The SIN validation is already present and the received status is different from
+          // the existing one. It can happen when the status received is 2 (under review) and later
+          // a new status will be received once the review is done, what means that one request can
+          // result in more than one response. In this case, if the received record is the most updated
+          // for the student, a "cloned" record will be inserted.
+          const mostUpdatedRecordDate = await this.repo
+            .createQueryBuilder("sinValidation")
+            .select("MAX(sinValidation.dateReceived)", "maxDate")
+            .innerJoin("sinValidation.student", "student")
+            .where("student.id = :student", {
+              student: existingValidation.student.id,
+            })
+            .getRawOne();
+          if (processDate > mostUpdatedRecordDate.maxDate) {
+            // This will force the record to be inserted with all the values selected
+            // during the initial SQL query.
+            delete existingValidation.id;
+            // Insert the new SIN validation and associate it with the student
+            // as the most updated one to be used as a current.
+            // A new SIN validation object created when below function
+            // is executed and updated in the same variable.
+            await this.studentService.updateSINValidationByStudentId(
```

Comment:
The `transactionalEntityManager` is provided as a parameter but it is not used to save the student. In there a reason for that?

Reply:
mm. it is 
![image](https://user-images.githubusercontent.com/77353155/211674664-3a5266fa-f705-43ab-9ebd-492c5aa9dee4.png)
. am I missing anything?

Reply:
nice catch